earth_enterprise/src/fusion/fusionui/QDateWrapper.cpp

```diff
@@ -130,6 +131,12 @@ void QDateWrapper::SetDate(const std::string& date) {
   }
 }
 
+bool QDateWrapper::IsValidDate() {
+  return QDate::isValid(year_edit_->text().toInt(),
```

Comment:
We can check if time is entered, using the TimeIncluded() function. If included: Time should also be validated here, using QDateTime:isValid .

Reply:
I wasn't sure if I should validate the time as well - it didn't seem like the user was able to input a valid time. With 00:00:00 being a valid time and each field being constrained to its upper bounds. Should the function I've created still validate the time as well?

Reply:
When I dug deeper, I found QDateWrapper.cpp already has :

void QDateWrapper::SetDateValidators(QObject* parent) {
 QIntValidator* year_validator = new QIntValidator(0, 9999, parent);
 QIntValidator* month_validator = new QIntValidator(0, 12, parent);
 QIntValidator* day_validator = new QIntValidator(0, 31, parent);

if we set the correct range here, then we may not need extra validations
this function prevents User from entering invalid values.

Reply:
I don't think that'd work for dates like it does for times since different months have a different number of days, especially with February and leap years.

Reply:
yes, you are right. we need the function you added for additional checking. But should we  also modify the SetDateValidators so that 0 is not a valid  entry ?